### ISSUE TITLE
Take into account tasks declaration order

### ISSUE THREAD

**Issue by vyshkant** [phpro/grumphp#509]
| Q               | A
| --------------- | ---
| Version         | `GrumPHP 0.14.0`
| Bug?            | no
| New feature?    | yes
| Question?       | no
| Documentation?  | no

When building a list of tasks, I propose to consider as the initial order the one in which they are declared in `grumphp.yml` (and than apply the priority sorting to this order).

Ideally I would like to see this feature by default, but if for some reasons it is unacceptable, I'd like to have a parameter like `respect_tasks_declaration_order` in `grumphp.yml`, which will enable the described behavior.

If you agree with my idea I'll gladly write a PR.

**Comment by Landerstraeten** [comment]
I don't understand your problem.

For what reason can't you use the metadata tag?

Example:
``` yaml
parameters:
  bin_dir: "./vendor/bin"
  git_dir: "."
  process_timeout: 120
  tasks:
      phpcs:
        standard: PSR2
        ignore_patterns:
          - "spec/*Spec.php"
          - "test/*.php"
        metadata:
          priority: 300
      phpspec:
        format: progress
        verbose: true
        metadata:
          priority: 200
      phpunit:
        metadata:
          priority: 100
      composer:
        no_check_lock: true
      yamllint:
        ignore_patterns:
          - "#test/(.*).yml#"
      phplint: ~
      phpparser:
        visitors:
          no_exit_statements: ~
          never_use_else: ~
          forbidden_function_calls:
            blacklist: [var_dump]
        metadata:
          priority: 100000
```

Result:
Running task 1/7: PhpParser
Running task 2/7: Phpcs
Running task 3/7: Phpspec
Running task 4/7: Phpunit
Running task 5/7: Composer
Running task 6/7: PhpLint
Running task 7/7: YamlLint

**Comment by vyshkant** [comment]
> For what reason can't you use the metadata tag?

Yes I can use the `metadata` tag. But for now the default order (when not using the `metadata` tag) is completely arbitrary from the user's point of view (the user does know anything about this order). And if tasks declaration order is taken into account, then the resulting order will be clear to the user.

**Comment by veewee** [comment]
Hi @vyshkant, 
It is simple to add, I'd accept a PR that adds this. Be sure to add tests and updated docs.

**Comment by veewee** [comment]
Fixed in #711
